src/Control/Effect/Fresh.hs

```diff
@@ -48,11 +48,10 @@ runFresh = fmap snd . flip runFreshC 0 . interpret
 newtype FreshC m a = FreshC { runFreshC :: Int -> m (Int, a) }
 
 instance (Carrier sig m, Effect sig, Monad m) => Carrier (Fresh :+: sig) (FreshC m) where
-  gen a = FreshC (\ i -> gen (i, a))
-  alg = algF \/ algOther
-    where algF (Fresh   k) = FreshC (\ i -> runFreshC (k i) (succ i))
-          algF (Reset m k) = FreshC (\ i -> runFreshC m i >>= \ (_, a) -> runFreshC (k a) i)
-          algOther op = FreshC (\ i -> alg (handle (i, ()) (uncurry (flip runFreshC)) op))
+  ret a = FreshC (\ i -> ret (i, a))
+  eff op = FreshC (\ i -> (alg i \/ eff . handle (i, ()) (uncurry (flip runFreshC))) op)
```

Comment:
Factoring `newtype` constructors out of the algebras cleans them up substantially.